Deeploy/TilingExtension/CodeTransformationPasses/TilingVariableReplacement.py

```diff
@@ -24,192 +24,206 @@
 # limitations under the License.
 
 import copy
-from typing import Dict, List, Tuple, Type
+import itertools
+from typing import List, Tuple
 
-from mako.parsetree import Expression, Node, Text
-
-from Deeploy.AbstractDataTypes import Pointer
+from Deeploy.AbstractDataTypes import Struct
 from Deeploy.CommonExtensions.CodeTransformationPasses.Closure import ClosureExecutionBlock
 from Deeploy.CommonExtensions.CodeTransformationPasses.IntrospectiveCodeTransformation import \
     IntrospectiveCodeTransformationMixIn
 from Deeploy.DeeployTypes import CodeGenVerbosity, CodeSnippet, CodeTransformationPass, ExecutionBlock, \
-    NetworkContext, NodeTemplate, OperatorRepresentation, TransientBuffer, _NoVerbosity
+    NetworkContext, NodeTemplate, OperatorRepresentation, TransientBuffer, VariableBuffer, _NoVerbosity, \
+    _ReferenceBuffer
+from Deeploy.TilingExtension.CodeTransformationPasses.TilingHoistingMixIn import TilingHoistingMixIn
 from Deeploy.TilingExtension.MemoryConstraints import NodeMemoryConstraint
 from Deeploy.TilingExtension.TilingCodegen import TilingSchedule, VariableReplacementScheme, minimizeVariableReplacement
 
 
-class TilingVariableReplacement(CodeTransformationPass, IntrospectiveCodeTransformationMixIn):
-
-    _prefix = "TILING_REPLACED_"
+class TilingVariableReplacement(CodeTransformationPass, IntrospectiveCodeTransformationMixIn, TilingHoistingMixIn):
 
     def __init__(self, targetMemLevel: str):
         self.targetMemLevel = targetMemLevel
-        self._name: str
+        TilingHoistingMixIn.__init__(self, targetMemLevel)
 
     @property
-    def prefix(self):
-        return self._prefix + f"{self._name}_" + self.targetMemLevel + "_"
+    def arenaName(self):
+        return f"MEMORYARENA_{self.targetMemLevel}"
```

Comment:
It seems like this might lead to problems with the definition in `Deeploy/TilingExtension/TilerExtension.py`.

You could use
```
return f"{Tiler.arenaName}_{self.targetMemLevel}"
```